schema.go

```diff
@@ -261,15 +261,51 @@ func (s *Schema) Reconstruct(value interface{}, row Row) error {
 		v = v.Elem()
 	}
 
-	columns := make([][]Value, len(s.columns))
+	b := valuesSliceBufferPool.Get().(*valuesSliceBuffer)
+
+	columns := b.reserve(len(s.columns))
 	row.Range(func(columnIndex int, columnValues []Value) bool {
 		if columnIndex < len(columns) {
 			columns[columnIndex] = columnValues
 		}
 		return true
 	})
+	// we avoid the defer penalty by releasing b manually
+	err := s.reconstruct(v, levels{}, columns)
+	b.release()
+	return err
+}
 
-	return s.reconstruct(v, levels{}, columns)
+type valuesSliceBuffer struct {
+	values [][]Value
+}
+
+func (v *valuesSliceBuffer) reserve(n int) [][]Value {
+	if n <= cap(v.values) {
+		return v.values[:n]
+	}
+	// we can try to keep growing by the power of two, but we care more about the
+	// memory footprint so  this should suffice.
+	//
+	// The nature of reads tends to be from similar number of columns.The less work
+	// we do here the better performance we can get.
+	v.values = make([][]Value, n)
```

Comment:
Since generic reader will have a fixed schema, why not have a pool per reader so that the size of the slice is known?

Reply:
>Since generic reader will have a fixed schema, why not have a pool per reader so that the size of the slice is known?

For all schemas with columns `<= 64` we get pre allocated slices from the pool so this is general improvement on majority of the cases.

This is an edge case where columns exceed `64`.  A simpler general solution in my opinion. Let's not optimise for edge cases.

I will keep this for now

Reply:
Right, but they are also over-allocated :)

Reply:
Isn't that the point?  All pooling strategies for allocations have classes which are usually a power of two ? Please see the comment where I explain why I chose `64`. 

You can argue for  `4`,`8` or `32` if the concern is `64` is too big. Something to note, the memory footprint is amortised  as supported by the benchmarks( btw thanks for the `-count` flag trick).